### PR TITLE
Input component

### PR DESCRIPTION
**Pull request by dzimmerman902**
## Guidelines

DEVELOPMENT branch on the LEFT <br>
YOUR branch on the RIGHT

## Issue

This issue resolves N/A

## Description

- Redid input component to specifically add or remove active class. This tells the label element to move up and down to give it a clean effect.
- Updated form component to handle validation. Requires validation prop (by itself) to determine if the form needs to be validated or not. 
- Created a new LoginForm component to handle all of the state for authorization instead of putting it all in the page
- Added ErrrorMessage component to help with handling some

### CODE REVIEW

File: client/src/components/LoginForm/LoginForm.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        history.push('/dashboard');
+      });
+    }
+  };
+
+  render() {
+    const { errorMessage } = this.props.auth;
+
+    return (
+      <Form handleFormSubmit={this.handleFormSubmit} validate>
```
[the remark this answers is not in the document]
Comment (dzimmerman902) [reply client/src/components/LoginForm/LoginForm.js]: It does not, good catch. I'll try to find a way to distinguish it

File: client/src/utils/validation/patterns.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 const patterns = {
-  can_delete: /^(true|false)$/,
+  canDelete: /^(true|false)$/,
   email: /^([a-zA-Z\d.-]+)@([a-zA-Z\d-]+\.)([a-zA-Z]{2,8})(.[a-zA-Z]{2,8})?$/,
-  first_name: /^[a-zA-Z0-9\s]{1,}$/,
-  last_name: /^[a-zA-Z0-9\s]{1,}$/,
-  on_air_name: /^[a-zA-Z0-9\s]{1,}$/,
+  firstName: /^[a-zA-Z0-9\s]{1,}$/,
```
[the remark this answers is not in the document]
Comment (dzimmerman902) [reply client/src/utils/validation/patterns.js]: I like that idea, I have plans to clean up the validation utils to make it more modular.

File: client/src/components/LoginForm/LoginForm.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            name="password"
+            onChange={this.handleInputChange}
+            type="password"
+            value={this.state.password}
+          />
+          <Label htmlFor="password">Password</Label>
+          <Feedback>Must be Valid Password</Feedback>
+        </FormGroup>
+
+        <Button type="primary">Sign In</Button>
```
[the remark this answers is not in the document]
Comment (dzimmerman902) [reply client/src/components/LoginForm/LoginForm.js]: I forgot about the "type" attribute, that might cause some confusion with the HTML so I might have to change it around. Right now it's used to determine which class to apply. I'll look into fixing that

File: client/src/components/LoginForm/LoginForm.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  };
+
+  render() {
+    const { errorMessage } = this.props.auth;
+
+    return (
+      <Form handleFormSubmit={this.handleFormSubmit} validate>
+        {errorMessage && <ErrorMessage message={errorMessage} />}
+        <FormGroup>
+          <Input
```
[the remark this answers is not in the document]
Comment (dzimmerman902) [reply client/src/components/LoginForm/LoginForm.js]: I really like that idea! Keep it clean and simple!

File: client/src/utils/validation/validate.js [hunk 5]
```diff
@@ -1,44 +1,31 @@
...
+      classList.add('invalid');
+      return (valid = false);
+    }
 
-      if (item.value === '') {
-        item.classList.add('invalid');
-        return (valid = false);
-      }
-    });
+    if (value === '') {
```
[the remark this answers is not in the document]
Comment (dzimmerman902) [reply client/src/utils/validation/validate.js]: Yeah, that's the tricky thing right now. It's all or nothing so I'm definitely open to ideas on how to have some optional. I'm going to look into the children prop to see what we can access from it.

File: client/src/components/LoginForm/LoginForm.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (dzimmerman902) [reply client/src/components/LoginForm/LoginForm.js]: Made an update, let me know what you think

File: client/src/components/LoginForm/LoginForm.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jberns) [reply client/src/components/LoginForm/LoginForm.js]: One thing we can do is change the current "type" to "color,"  that way we can use the native HTML attributes.

File: client/src/components/LoginForm/LoginForm.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (jberns) [reply client/src/components/LoginForm/LoginForm.js]: Looks super clean!  I like it!

File: client/src/components/LoginForm/LoginForm.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (dzimmerman902) [reply client/src/components/LoginForm/LoginForm.js]: Read my mind

File: client/src/utils/validation/patterns.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dzimmerman902) [reply client/src/utils/validation/patterns.js]: Updated!

**Comment by jberns** [comment]
Thanks Ill take a more in depth look at it over the weekend.  Only items I noticed at first glance the Alert/Form components don't look as generic as they were.  And then how do we handle multiple forms?  I think we need to be able to name them and then get the right data using some sort of selector.

**Comment by dzimmerman902** [comment]
Sure thing, if you want to do a quick video call, I can explain it better!

**Pull request merged**